tests/test_dataapi.py

```diff
@@ -0,0 +1,138 @@
+        responses.add(responses.POST, endpoint, json={
+            'meta': {
+                'status': True,
+                'timestamp': 1514874527,
+                'records': 0,
+                'next': '1'
+            },
+            'data': [{'id': 'a'}]
+        })
+        responses.add(responses.POST, endpoint, json={
+            'meta': {
+                'status': True,
+                'timestamp': 1514874527,
+                'records': 0
+            },
+            'data': [{'id': 'b'}]
+        })
+        client = DataApi()
+        pages = client.request_iter(endpoint, security, consumer_secret,
+                                    request, action)
+        results = []
+        for page in pages:
+            results.append(page)
+
+        assert len(results) == 2
+        assert results[0]['data'][0]['id'] == 'a'
+        assert results[1]['data'][0]['id'] == 'b'
+
+    @responses.activate
+    def test_result_iter(self):
+        """Verify that `result_iter` returns an iterator of results"""
+        responses.add(responses.POST, endpoint, json={
+            'meta': {
+                'status': True,
+                'timestamp': 1514874527,
+                'records': 0,
+                'next': '1'
+            },
+            'data': [{'id': 'a'}]
+        })
+        responses.add(responses.POST, endpoint, json={
+            'meta': {
+                'status': True,
+                'timestamp': 1514874527,
+                'records': 0
+            },
+            'data': [{'id': 'b'}]
+        })
+        client = DataApi()
+        result_iter = client.results_iter(endpoint, security, consumer_secret,
+                                          request, action)
+        results = list(result_iter)
+
+        assert len(results) == 2
+        assert results[0]['id'] == 'a'
+        assert results[1]['id'] == 'b'
+
+
+class IntegrationTestDataApiClient(unittest.TestCase):
+
+    def test_real_request(self):
+        """Make a request against Data Api to ensure the SDK works"""
+        client = DataApi()
+        res = client.request(endpoint, security, consumer_secret, request,
+                             action)
+        print(res.request.url)
+        print(res.request.body)
+        print(res.content)
+        returned_json = res.json()
+        assert len(returned_json['data']) > 0
+        returned_ref = returned_json['data'][0]['reference']
+        assert returned_ref in request['references']
+
+    def test_paging(self):
+        """Verify that paging works"""
+        client = DataApi()
+        pages = client.request_iter(endpoint, security, consumer_secret,
+                                    request, action)
+        results = set()
+        for page in pages:
+            if page['data']:
+                results.add(page['data'][0]['reference'])
+
+        assert len(results) == 2
+        assert results == {'item_2', 'item_3'}
+
```

Comment:
Should we test `results_iter`, too?

Reply:
Don't we already have a test for that?

Reply:
Not in integration tests, I don't think.